### ISSUE TITLE
User Guide

### ISSUE THREAD

**Issue by danielecook** [samtools/htslib#114]
Is there a user guide / documentation for htslib?

**Comment by rulixxx** [comment]
Started using htslib myself some months ago and still find it difficult to find usage of some parts of it. It makes development more time consuming and error prone.

**Comment by valeriuo** [comment]
@rulixxx We are quite aware of this aspect and we are planning to invest more resources in the documentation and the user guide. In the meantime, can you describe your use case and see how we can assist you?
You can use samtools-help@lists.sourceforge.net, if you don't want to post it here.

**Comment by rulixxx** [comment]
Yes I was aware that this is a common complaint of users. I just wanted to add my grain of sand to addressing the issue (not just complain and forget about it).

Thanks for your offer Valeriu. Thus far I have been able to get by looking at examples, header files and generated pages. My most recent problem was hunting down the return codes for several functions from htslib in order to harden a piece of code. I had to jump from headers to various documents in order to get the full picture for each function (not really sure Im looking at the most adequate/ updated source in the web). I just thought it would save time having all of it in one place with a simple navigation inferface...


________________________________
